File: aura/src/services/analyticsService.ts
/**
 * Analytics Service
 * 
 * Revenue-optimized analytics service that drives strategic upgrades
 * through intelligent data presentation and value demonstration.
 */

import { apiService } from './apiService';
import { demoService } from './demoService';
import type { UsageMetrics, UserSubscription } from '../types/pricing';

export interface DashboardStats {
  subscription: UserSubscription;
  usage_metrics: UsageMetrics;
  recent_activities: Activity[];
  upgrade_opportunities: UpgradeOpportunity[];
  success_metrics: SuccessMetric[];
}

export interface Activity {
  id: string;
  type: 'parse' | 'plan' | 'execute' | 'system';
  title: string;
  description: string;
  timestamp: string;
  status: 'success' | 'warning' | 'error' | 'info';
  metadata?: Record<string, any>;
}

export interface UpgradeOpportunity {
  id: string;
  type: 'usage_limit' | 'feature_unlock' | 'value_demonstration';
  title: string;
  description: string;
  urgency: 'low' | 'medium' | 'high';
  cta_text: string;
  cta_action: string;
  value_props: string[];
  estimated_roi?: string;
}

export interface SuccessMetric {
  label: string;
  value: string;
  trend: string;
  trend_direction: 'up' | 'down' | 'stable';
  benchmark?: string;
  tooltip?: string;
}

class AnalyticsService {
  
  /**
   * Get comprehensive dashboard statistics with revenue optimization
   */
  async getDashboardStats(): Promise<DashboardStats> {
    try {
      if (demoService.isDemoMode()) {
        console.log('🎭 Demo Mode: Using analytics demo data');
        return this.getDemoAnalytics();
      }

      // Production analytics - parallel API calls for performance
      const [subscription, usageMetrics, activities] = await Promise.all([
        this.getUserSubscription(),
        this.getUsageMetrics(),
        this.getRecentActivities()
      ]);

      const upgradeOpportunities = this.generateUpgradeOpportunities(subscription, usageMetrics);
      const successMetrics = this.calculateSuccessMetrics(usageMetrics);

      return {
        subscription,
        usage_metrics: usageMetrics,
        recent_activities: activities,
        upgrade_opportunities: upgradeOpportunities,
        success_metrics: successMetrics
      };

    } catch (error) {
      console.error('Failed to load dashboard stats:', error);
      throw error;
    }
  }

  /**
   * Get user's current subscription details
   */
  async getUserSubscription(): Promise<UserSubscription> {
    return apiService.get<UserSubscription>('/analytics/subscription');
  }

  /**
   * Get detailed usage metrics across all platform components
   */
  async getUsageMetrics(hours: number = 24): Promise<UsageMetrics> {
    return apiService.get<UsageMetrics>(`/analytics/usage?hours=${hours}`);
  }

  /**
   * Get comprehensive analytics dashboard data
   */
  async getAnalyticsDashboard(): Promise<DashboardStats> {
    return apiService.get<DashboardStats>('/analytics/dashboard');
  }

  /**
   * Get strategic upgrade opportunities
   */
  async getUpgradeOpportunities(): Promise<UpgradeOpportunity[]> {
    return apiService.get<UpgradeOpportunity[]>('/analytics/upgrade-opportunities');
  }

  /**
   * Get component-specific metrics
   */
  async getComponentMetrics(component: string, hours: number = 24): Promise<any> {
    return apiService.get(`/analytics/component/${component}?hours=${hours}`);
  }

  /**
   * Get ROI calculation
   */
  async getROICalculation(hourlyRate: number = 50): Promise<any> {
    return apiService.get(`/analytics/roi-calculation?hourly_rate=${hourlyRate}`);
  }

  /**
   * Get success metrics for value demonstration
   */
  async getSuccessMetrics(): Promise<SuccessMetric[]> {
    return apiService.get<SuccessMetric[]>('/analytics/success-metrics');
  }

  /**
   * Get billing information
   */
  async getBillingInfo(): Promise<any> {
    return apiService.get('/analytics/billing');
  }

  /**
   * Track analytics events for conversion optimization
   */
  async trackEvent(eventType: string, eventData: any): Promise<void> {
    return apiService.post('/analytics/track-event', {
      event_type: eventType,
      event_data: eventData
    });
  }

  /**
   * Get recent activities with intelligent categorization
   */
  async getRecentActivities(): Promise<Activity[]> {
    return apiService.get<Activity[]>('/analytics/activities?limit=10');
  }

  /**
   * Get Reader (parsing) specific analytics
   */
  async getReaderAnalytics(timeframe: '7d' | '30d' | '90d' = '30d') {
    return apiService.get(`/analytics/reader?timeframe=${timeframe}`);
  }

  /**
   * Get Planner (AI planning) specific analytics  
   */
  async getPlannerAnalytics(timeframe: '7d' | '30d' | '90d' = '30d') {
    return apiService.get(`/analytics/planner?timeframe=${timeframe}`);
  }

  /**
   * Get Actor (execution) specific analytics
   */
  async getActorAnalytics(timeframe: '7d' | '30d' | '90d' = '30d') {
    return apiService.get(`/analytics/actor?timeframe=${timeframe}`);
  }

  /**
   * Get unified cross-platform analytics (Complete/Enterprise only)
   */
  async getUnifiedAnalytics(timeframe: '7d' | '30d' | '90d' = '30d') {
    return apiService.get(`/analytics/unified?timeframe=${timeframe}`);
  }

  /**
   * Get ROI and business impact metrics
   */
  async getROIMetrics() {
    return apiService.get('/analytics/roi');
  }

  /**
   * Generate intelligent upgrade opportunities based on usage patterns
   */
  private generateUpgradeOpportunities(
    subscription: UserSubscription, 
    metrics: UsageMetrics
  ): UpgradeOpportunity[] {
    const opportunities: UpgradeOpportunity[] = [];

    // Usage-based upgrade prompts
    if (subscription.tier === 'free') {
      const parseUsage = subscription.usage.parses_used / (subscription.limits.parses as number);
      const planUsage = subscription.usage.plans_used / (subscription.limits.plans as number);
      const execUsage = subscription.usage.executions_used / (subscription.limits.executions as number);

      if (parseUsage > 0.8 || planUsage > 0.8 || execUsage > 0.8) {
        opportunities.push({
          id: 'free_tier_limits',
          type: 'usage_limit',
          title: 'Approaching Free Tier Limits',
          description: `You've used ${Math.max(parseUsage, planUsage, execUsage) * 100}% of your allocation. Upgrade to avoid interruptions.`,
          urgency: 'high',
          cta_text: 'Upgrade to Complete Platform',
          cta_action: 'upgrade_complete',
          value_props: [
            'Unlimited everything for $399/mo',
            '40% savings vs individual tools',
            'Advanced analytics included'
          ],
          estimated_roi: 'Users save 15+ hours/week on average'
        });
      }

      // Value demonstration for free users
      if (metrics.unified.total_workflows > 5) {
        opportunities.push({
          id: 'success_demonstration',
          type: 'value_demonstration',
          title: 'Unlock Your Full Potential',
          description: 'You\'ve completed 5+ workflows! See how Complete Platform users achieve 97% automation success.',
          urgency: 'medium',
          cta_text: 'See Success Stories',
          cta_action: 'view_success_stories',
          value_props: [
            '3x better automation ROI',
            'Advanced workflow optimization',
            'Unified cross-tool insights'
          ]
        });
      }
    }

    // Individual tool users -> Bundle upgrade
    if (['reader_pro', 'planner_pro', 'actor_pro'].includes(subscription.tier)) {
      opportunities.push({
        id: 'bundle_savings',
        type: 'feature_unlock',
        title: 'Save 40% with Complete Platform',
        description: 'Unlock all tools + unified analytics for less than what you\'re paying now.',
        urgency: 'medium',
        cta_text: 'Calculate Savings',
        cta_action: 'show_bundle_calculator',
        value_props: [
          'Access all WebAgent tools',
          'Unified workflow analytics',
          'Save $138/month vs current plan'
        ],
        estimated_roi: '$1,656 annual savings'
      });
    }

    return opportunities;
  }

  /**
   * Calculate success metrics that demonstrate platform value
   */
  private calculateSuccessMetrics(metrics: UsageMetrics): SuccessMetric[] {
    return [
      {
        label: 'Automation Success Rate',
        value: `${Math.round(metrics.unified.end_to_end_success_rate * 100)}%`,
        trend: '+5%',
        trend_direction: 'up',
        benchmark: 'Industry avg: 73%',
        tooltip: 'Percentage of workflows completed successfully end-to-end'
      },
      {
        label: 'Time Saved This Month',
        value: `${Math.round(metrics.unified.roi_metrics.time_saved_hours)} hours`,
        trend: '+12 hours',
        trend_direction: 'up',
        tooltip: 'Total hours saved through automation vs manual processes'
      },
      {
        label: 'Cost Savings Generated',
        value: `$${Math.round(metrics.unified.roi_metrics.cost_saved_usd).toLocaleString()}`,
        trend: '+$1,250',
        trend_direction: 'up',
        tooltip: 'Estimated cost savings from automated workflows'
      },
      {
        label: 'User Satisfaction',
        value: `${Math.round(metrics.unified.user_satisfaction_score * 100)}%`,
        trend: 'stable',
        trend_direction: 'stable',
        benchmark: 'Enterprise avg: 94%'
      }
    ];
  }

  /**
   * Demo mode analytics data
   */
  private async getDemoAnalytics(): Promise<DashboardStats> {
    const now = new Date();
    const thirtyDaysAgo = new Date(now.getTime() - 30 * 24 * 60 * 60 * 1000);

    return {
      subscription: {
        tier: 'free',
        status: 'active',
        current_period_start: thirtyDaysAgo.toISOString(),
        current_period_end: new Date(now.getTime() + 5 * 24 * 60 * 60 * 1000).toISOString(),
        usage: {
          parses_used: 87,
          plans_used: 12,
          executions_used: 6,
          storage_used_gb: 0.3,
          api_calls_used: 234
        },
        limits: {
          parses: 200,
          plans: 20,
          executions: 10,
          storage_gb: 1,
          api_calls: 1000
        }
      },
      usage_metrics: {
        reader: {
          total_parses: 87,
          successful_parses: 82,
          failed_parses: 5,
          success_rate: 0.94,
          avg_parse_time_ms: 1250,
          domains_parsed: ['github.com', 'stackoverflow.com', 'docs.python.org'],
          performance_trend: [
            { date: '2025-06-15', value: 1.2 },
            { date: '2025-06-16', value: 1.3 },
            { date: '2025-06-17', value: 1.1 },
            { date: '2025-06-18', value: 1.25 },
            { date: '2025-06-19', value: 1.2 },
            { date: '2025-06-20', value: 1.25 }
          ]
        },
        planner: {
          total_plans: 12,
          successful_plans: 11,
          failed_plans: 1,
          avg_confidence_score: 0.87,
          avg_plan_complexity: 6.4,
          plan_types: [
            { type: 'web_scraping', count: 5 },
            { type: 'form_filling', count: 4 },
            { type: 'data_extraction', count: 3 }
          ],
          confidence_trend: [
            { date: '2025-06-15', value: 0.85 },
            { date: '2025-06-16', value: 0.88 },
            { date: '2025-06-17', value: 0.86 },
            { date: '2025-06-18', value: 0.89 },
            { date: '2025-06-19', value: 0.87 },
            { date: '2025-06-20', value: 0.87 }
          ]
        },
        actor: {
          total_executions: 6,
          successful_executions: 6,
          failed_executions: 0,
          success_rate: 1.0,
          avg_execution_time_ms: 45000,
          actions_per_execution: 12.5,
          error_categories: [],
          performance_trend: [
            { date: '2025-06-18', value: 42 },
            { date: '2025-06-19', value: 48 },
            { date: '2025-06-20', value: 45 }
          ]
        },
        unified: {
          total_workflows: 6,
          end_to_end_success_rate: 0.92,
          avg_workflow_time_ms: 180000,
          roi_metrics: {
            time_saved_hours: 18.5,
            cost_saved_usd: 2750,
            automation_value: 4200
          },
          user_satisfaction_score: 0.95
        }
      },
      recent_activities: [
        {
          id: '1',
          type: 'execute',
          title: 'Workflow Completed Successfully',
          description: 'Data extraction from GitHub repositories completed',
          timestamp: new Date(now.getTime() - 2 * 60 * 60 * 1000).toISOString(),
          status: 'success',
          metadata: { execution_time: '2m 15s', actions: 8 }
        },
        {
          id: '2', 
          type: 'plan',
          title: 'New Automation Plan Created',
          description: 'AI planning completed for form filling workflow',
          timestamp: new Date(now.getTime() - 4 * 60 * 60 * 1000).toISOString(),
          status: 'success',
          metadata: { confidence: 0.89, complexity: 7 }
        },
        {
          id: '3',
          type: 'parse',
          title: 'Website Parsing Completed',
          description: 'Extracted 45 data points from stackoverflow.com',
          timestamp: new Date(now.getTime() - 6 * 60 * 60 * 1000).toISOString(),
          status: 'success',
          metadata: { elements_found: 45, parse_time: '1.2s' }
        }
      ],
      upgrade_opportunities: [
        {
          id: 'demo_upgrade',
          type: 'value_demonstration',
          title: 'You\'re Doing Great! 🚀',
          description: 'You\'ve automated 6 workflows successfully. See how Complete Platform users achieve 97% success rates.',
          urgency: 'medium',
          cta_text: 'Explore Complete Platform',
          cta_action: 'view_complete_features',
          value_props: [
            'Unlimited everything for $399/mo',
            'Advanced cross-tool analytics',
            '40% savings vs individual tools'
          ],
          estimated_roi: 'Users typically save 15+ hours/week'
        }
      ],
      success_metrics: [
        {
          label: 'Success Rate',
          value: '92%',
          trend: '+8%',
          trend_direction: 'up',
          benchmark: 'Your best yet!'
        },
        {
          label: 'Time Saved',
          value: '18.5 hours',
          trend: '+6 hours',
          trend_direction: 'up'
        },
        {
          label: 'Cost Savings',
          value: '$2,750',
          trend: '+$850',
          trend_direction: 'up'
        },
        {
          label: 'Satisfaction',
          value: '95%',
          trend: 'stable',
          trend_direction: 'stable'
        }
      ]
    };
  }
}

export const analyticsService = new AnalyticsService();
export default analyticsService;

File: app/services/analytics_service.py
"""
Analytics Service

Revenue-optimized analytics service for WebAgent's 2025 pricing model.
Implements intelligent upgrade detection and conversion optimization.
"""

from datetime import datetime, timedelta
from typing import Any

import structlog
from sqlalchemy import and_, func, select
from sqlalchemy.ext.asyncio import AsyncSession

from app.models.execution_plan import ExecutionPlan
from app.models.task import Task
from app.schemas.analytics import (
    AnalyticsDashboard,
    ComponentMetrics,
    ComponentType,
    ConversionMetrics,
    PlatformAnalytics,
    ROICalculation,
    SubscriptionTier,
    SuccessMetric,
    TrendDirection,
    UpgradeOpportunity,
    UsageMetrics,
)
from app.services.billing_service import BillingService
from app.services.subscription_service import SubscriptionService

logger = structlog.get_logger(__name__)


class AnalyticsService:
    """
    Analytics service for revenue-optimized dashboard and conversion tracking.
    """

    def __init__(self):
        self.subscription_service = SubscriptionService()
        self.billing_service = BillingService()

    async def get_dashboard_data(
        self, db: AsyncSession, user_id: int
    ) -> AnalyticsDashboard:
        """
        Get comprehensive analytics dashboard data with revenue optimization.
        """

        logger.info("Generating analytics dashboard", user_id=user_id)

        # Get user subscription and billing info
        subscription = await self.subscription_service.get_user_subscription(
            db, user_id
        )
        billing_info = await self.billing_service.get_billing_info(db, user_id)

        # Get usage metrics
        usage_metrics = await self.get_usage_metrics(db, user_id, hours=24)

        # Get platform analytics
        platform_analytics = await self.get_platform_analytics(db, user_id)

        # Generate success metrics for value demonstration
        success_metrics = await self.get_success_metrics(db, user_id)

        # Calculate ROI for value proposition
        roi_calculation = await self.calculate_roi(db, user_id)

        # Generate strategic upgrade opportunities
        upgrade_opportunities = await self.get_upgrade_opportunities(db, user_id)

        # Get conversion metrics
        conversion_metrics = await self.get_conversion_metrics(db, user_id)

        # Set cache expiry
        cache_expires_at = datetime.utcnow() + timedelta(minutes=5)

        return AnalyticsDashboard(
            user_id=user_id,
            subscription=subscription,
            usage_metrics=usage_metrics,
            platform_analytics=platform_analytics,
            success_metrics=success_metrics,
            roi_calculation=roi_calculation,
            upgrade_opportunities=upgrade_opportunities,
            conversion_metrics=conversion_metrics,
            billing_info=billing_info,
            cache_expires_at=cache_expires_at,
        )

    async def get_usage_metrics(
        self, db: AsyncSession, user_id: int, hours: int = 24
    ) -> UsageMetrics:
        """
        Get comprehensive usage metrics for the specified time period.
        """

        cutoff_time = datetime.utcnow() - timedelta(hours=hours)

        # Get task counts (parsing operations)
        parse_query = select(func.count(Task.id)).where(
            and_(Task.user_id == user_id, Task.created_at >= cutoff_time)
        )
        parses_used = (await db.execute(parse_query)).scalar() or 0

        # Get execution plan counts (planning operations)
        plan_query = select(func.count(ExecutionPlan.id)).where(
            and_(
                ExecutionPlan.user_id == user_id,
                ExecutionPlan.created_at >= cutoff_time,
            )
        )
        plans_used = (await db.execute(plan_query)).scalar() or 0

        # Get execution counts (action operations)
        execution_query = select(func.count(ExecutionPlan.id)).where(
            and_(
                ExecutionPlan.user_id == user_id,
                ExecutionPlan.status == "completed",
                ExecutionPlan.created_at >= cutoff_time,
            )
        )
        executions_used = (await db.execute(execution_query)).scalar() or 0

        # Calculate success rates
        total_tasks = (
            await db.execute(select(func.count(Task.id)).where(Task.user_id == user_id))
        ).scalar() or 0

        successful_tasks = (
            await db.execute(
                select(func.count(Task.id)).where(
                    and_(Task.user_id == user_id, Task.status == "completed")
                )
            )
        ).scalar() or 0

        parse_success_rate = (
            (successful_tasks / total_tasks * 100) if total_tasks > 0 else 0
        )

        # Generate trend data
        daily_usage = await self._generate_daily_usage_trend(db, user_id, hours)

        # Calculate performance metrics
        avg_parse_time = await self._calculate_avg_parse_time(db, user_id)
        avg_plan_time = await self._calculate_avg_plan_time(db, user_id)
        avg_execution_time = await self._calculate_avg_execution_time(db, user_id)

        # Calculate ROI metrics
        time_saved_hours = parses_used * 0.5 + plans_used * 1.0 + executions_used * 2.0
        cost_saved_usd = time_saved_hours * 50.0  # Assume $50/hour
        automation_value = cost_saved_usd * 1.2  # 20% efficiency bonus

        return UsageMetrics(
            parses_used=parses_used,
            plans_used=plans_used,
            executions_used=executions_used,
            api_calls_used=parses_used + plans_used + executions_used,
            storage_used_gb=0.1 * (parses_used + plans_used),  # Estimate
            daily_usage=daily_usage,
            weekly_trends=[],
            monthly_summary={},
            avg_parse_time_ms=avg_parse_time,
            avg_plan_time_ms=avg_plan_time,
            avg_execution_time_ms=avg_execution_time,
            parse_success_rate=parse_success_rate,
            plan_success_rate=85.0,  # Default high success rate
            execution_success_rate=90.0,  # Default high success rate
            time_saved_hours=time_saved_hours,
            cost_saved_usd=cost_saved_usd,
            automation_value=automation_value,
        )

    async def get_upgrade_opportunities(
        self, db: AsyncSession, user_id: int
    ) -> list[UpgradeOpportunity]:
        """
        Generate strategic upgrade opportunities for conversion optimization.
        """

        opportunities = []
        subscription = await self.subscription_service.get_user_subscription(
            db, user_id
        )
        usage_metrics = await self.get_usage_metrics(db, user_id, hours=720)  # 30 days

        # Usage-based upgrade opportunities
        if subscription.tier == SubscriptionTier.FREE:
            # Check if approaching limits
            parse_usage_pct = (usage_metrics.parses_used / 200) * 100
            plan_usage_pct = (usage_metrics.plans_used / 20) * 100
            execution_usage_pct = (usage_metrics.executions_used / 10) * 100

            if parse_usage_pct > 80:
                opportunities.append(
                    UpgradeOpportunity(
                        type="usage_limit",
                        priority=1,
                        title="Reader Usage Approaching Limit",
                        description=f"You've used {usage_metrics.parses_used}/200 website parses this month",
                        value_proposition="Upgrade to Reader Pro for unlimited parsing + advanced analytics",
                        current_tier=subscription.tier,
                        recommended_tier=SubscriptionTier.READER_PRO,
                        savings_amount=0,
                        savings_percentage=0,
                        cta_text="Upgrade to Reader Pro - $129/mo",
                        usage_percentage=parse_usage_pct,
                        days_until_limit=max(
                            1,
                            int(
                                (200 - usage_metrics.parses_used)
                                / max(1, usage_metrics.parses_used / 30)
                            ),
                        ),
                    )
                )

            # Bundle savings opportunity
            if parse_usage_pct > 60 and plan_usage_pct > 60:
                opportunities.append(
                    UpgradeOpportunity(
                        type="savings",
                        priority=2,
                        title="Complete Platform - 40% Savings",
                        description="You're using multiple AI components - save with our bundle",
                        value_proposition="Get Reader + Planner + Actor for $399/mo (save $138/mo vs individual)",
                        current_tier=subscription.tier,
                        recommended_tier=SubscriptionTier.COMPLETE_PLATFORM,
                        savings_amount=138,
                        savings_percentage=40,
                        cta_text="Save $138/mo with Complete Platform",
                        usage_percentage=max(parse_usage_pct, plan_usage_pct),
                    )
                )

        # Feature unlock opportunities
        if subscription.tier in [SubscriptionTier.FREE, SubscriptionTier.READER_PRO]:
            opportunities.append(
                UpgradeOpportunity(
                    type="feature_unlock",
                    priority=3,
                    title="Unlock AI Planning Intelligence",
                    description="Transform goals into executable plans with our AI Brain",
                    value_proposition="Enterprise customers achieve 3x automation ROI with AI planning",
                    current_tier=subscription.tier,
                    recommended_tier=SubscriptionTier.PLANNER_PRO,
                    cta_text="Unlock AI Planning - $179/mo",
                )
            )

        # Performance boost opportunities
        if usage_metrics.avg_parse_time_ms > 5000:  # Slow performance
            opportunities.append(
                UpgradeOpportunity(
                    type="performance_boost",
                    priority=4,
                    title="Priority Processing Available",
                    description="Reduce wait times with priority queue access",
                    value_proposition="Pro users get 3x faster processing and priority support",
                    current_tier=subscription.tier,
                    recommended_tier=SubscriptionTier.COMPLETE_PLATFORM,
                    cta_text="Get Priority Processing",
                )
            )

        return opportunities

    async def get_success_metrics(
        self, db: AsyncSession, user_id: int
    ) -> list[SuccessMetric]:
        """
        Generate success metrics for value demonstration.
        """

        usage_metrics = await self.get_usage_metrics(db, user_id)

        metrics = [
            SuccessMetric(
                label="Automation Success Rate",
                value=f"{usage_metrics.parse_success_rate:.1f}%",
                trend=2.5,
                trend_direction=TrendDirection.UP,
                benchmark="Industry avg: 73%",
                tooltip="Percentage of successful automation tasks",
            ),
            SuccessMetric(
                label="Time Saved This Month",
                value=f"{usage_metrics.time_saved_hours:.1f} hours",
                trend=15.2,
                trend_direction=TrendDirection.UP,
                benchmark="Pro users avg: 45 hours",
                tooltip="Total time saved through automation",
            ),
            SuccessMetric(
                label="Cost Savings",
                value=f"${usage_metrics.cost_saved_usd:.0f}",
                trend=8.7,
                trend_direction=TrendDirection.UP,
                benchmark="Enterprise avg: $2,400",
                tooltip="Estimated labor cost savings",
            ),
            SuccessMetric(
                label="Automation Value",
                value=f"${usage_metrics.automation_value:.0f}",
                trend=12.3,
                trend_direction=TrendDirection.UP,
                benchmark="Complete Platform avg: $3,200",
                tooltip="Total value generated through automation",
            ),
        ]

        return metrics

    async def calculate_roi(
        self, db: AsyncSession, user_id: int, hourly_rate: float = 50.0
    ) -> ROICalculation:
        """
        Calculate comprehensive ROI metrics for value demonstration.
        """

        usage_metrics = await self.get_usage_metrics(db, user_id, hours=720)  # 30 days
        subscription = await self.subscription_service.get_user_subscription(
            db, user_id
        )

        # Calculate time saved
        time_saved_hours = usage_metrics.time_saved_hours

        # Calculate cost savings
        labor_cost_saved = time_saved_hours * hourly_rate

        # Get subscription cost
        subscription_cost = subscription.monthly_cost

        # Calculate net savings and ROI
        net_savings = labor_cost_saved - subscription_cost
        roi_percentage = (
            (net_savings / max(subscription_cost, 1)) * 100
            if subscription_cost > 0
            else 0
        )

        # Calculate payback period
        daily_savings = labor_cost_saved / 30
        payback_period_days = (
            int(subscription_cost / max(daily_savings, 1)) if daily_savings > 0 else 0
        )

        # Annual value projection
        annual_value = net_savings * 12

        return ROICalculation(
            time_saved_hours=time_saved_hours,
            hourly_rate_usd=hourly_rate,
            labor_cost_saved=labor_cost_saved,
            subscription_cost=subscription_cost,
            net_savings=net_savings,
            roi_percentage=roi_percentage,
            payback_period_days=payback_period_days,
            annual_value=annual_value,
        )

    async def _generate_daily_usage_trend(
        self, db: AsyncSession, user_id: int, hours: int
    ) -> list[dict[str, Any]]:
        """Generate daily usage trend data."""
        # Implementation would query database for daily usage patterns
        # For now, return sample data
        return [
            {"date": "2025-06-19", "parses": 15, "plans": 3, "executions": 2},
            {"date": "2025-06-20", "parses": 22, "plans": 5, "executions": 4},
        ]

    async def _calculate_avg_parse_time(self, db: AsyncSession, user_id: int) -> float:
        """Calculate average parsing time."""
        # Implementation would calculate from task execution times
        return 2500.0  # Sample: 2.5 seconds

    async def _calculate_avg_plan_time(self, db: AsyncSession, user_id: int) -> float:
        """Calculate average planning time."""
        return 8500.0  # Sample: 8.5 seconds

    async def _calculate_avg_execution_time(
        self, db: AsyncSession, user_id: int
    ) -> float:
        """Calculate average execution time."""
        return 15000.0  # Sample: 15 seconds

    async def get_platform_analytics(
        self, db: AsyncSession, user_id: int
    ) -> PlatformAnalytics:
        """Get comprehensive platform analytics."""
        # Implementation would aggregate metrics across all components

        reader_metrics = ComponentMetrics(
            component=ComponentType.READER,
            total_requests=100,
            successful_requests=95,
            success_rate=95.0,
            avg_response_time_ms=2500.0,
        )

        planner_metrics = ComponentMetrics(
            component=ComponentType.PLANNER,
            total_requests=50,
            successful_requests=47,
            success_rate=94.0,
            avg_response_time_ms=8500.0,
        )

        actor_metrics = ComponentMetrics(
            component=ComponentType.ACTOR,
            total_requests=25,
            successful_requests=23,
            success_rate=92.0,
            avg_response_time_ms=15000.0,
        )

        unified_metrics = ComponentMetrics(
            component=ComponentType.UNIFIED,
            total_requests=25,
            successful_requests=22,
            success_rate=88.0,
            avg_response_time_ms=26000.0,
        )

        return PlatformAnalytics(
            reader_metrics=reader_metrics,
            planner_metrics=planner_metrics,
            actor_metrics=actor_metrics,
            unified_metrics=unified_metrics,
            workflow_success_rate=88.0,
            end_to_end_performance=85.0,
            integration_health=92.0,
            feature_adoption_rates={
                "reader": 0.95,
                "planner": 0.75,
                "actor": 0.45,
                "unified": 0.35,
            },
            user_journey_analytics={},
            engagement_metrics={},
        )

    async def get_conversion_metrics(
        self, db: AsyncSession, user_id: int
    ) -> ConversionMetrics:
        """Get conversion optimization metrics."""

        return ConversionMetrics(
            usage_based_upgrades=0,
            feature_based_upgrades=0,
            savings_based_upgrades=0,
            individual_to_bundle_rate=0.55,
            bundle_to_enterprise_rate=0.25,
            addon_attachment_rate=0.40,
            demo_to_conversion_rate=0.35,
            trial_engagement_score=0.75,
        )

    async def get_component_metrics(
        self, db: AsyncSession, user_id: int, component_type: ComponentType, hours: int
    ) -> ComponentMetrics:
        """Get detailed metrics for specific component."""
        # Implementation would query component-specific metrics
        return ComponentMetrics(
            component=component_type,
            total_requests=100,
            successful_requests=95,
            success_rate=95.0,
            avg_response_time_ms=2500.0,
            accuracy_score=0.92,
            confidence_score=0.88,
            user_satisfaction=0.91,
        )

    async def track_event(
        self, db: AsyncSession, user_id: int, event_type: str, event_data: dict
    ):
        """Track analytics events for conversion optimization."""
        logger.info(
            "Tracking analytics event",
            user_id=user_id,
            event_type=event_type,
            event_data=event_data,
        )
        # Implementation would store event in analytics database
